src/reader.rs

```diff
@@ -27,33 +33,50 @@ use super::utils;
 /// an `io::ErrorKind::TimedOut` variant as the value of `io::Error`. All other
 /// error values that would normally be produced by the underlying implementation
 /// of the `Read` trait could also be produced by the `TimeoutReader`.
-pub struct TimeoutReader<H>
-where
-    H: Read + AsRawFd,
-{
+pub struct TimeoutReader<H> {
     timeout: Option<c_int>,
     handle: H,
 }
 
+#[cfg(unix)]
 impl<H> Read for TimeoutReader<H>
 where
     H: Read + AsRawFd,
 {
     fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
-        utils::wait_until_ready(self.timeout, &self.handle, PollFlags::POLLIN)?;
+        utils::wait_until_ready(self.timeout,
+            &self.handle, PollFlags::POLLIN)?;
+        self.handle.read(buf)
+    }
+}
+
+#[cfg(windows)]
+impl<H> Read for TimeoutReader<H>
+where
+    H: Read + AsRawHandle,
+{
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
+        if let Some(timeout) = self.timeout {
+            let handle = self.handle.as_raw_handle();
+            let mut timeouts = unsafe { ::std::mem::zeroed::<um::winbase::COMMTIMEOUTS>() };
+            timeouts.ReadTotalTimeoutConstant = timeout as u32;
+
+            unsafe { um::commapi::SetCommTimeouts(handle, &mut timeouts) };
+        }
         self.handle.read(buf)
     }
 }
 
 impl<H> Seek for TimeoutReader<H>
 where
-    H: Read + AsRawFd + Seek,
+    H: Seek,
 {
     fn seek(&mut self, pos: SeekFrom) -> Result<u64> {
         self.handle.seek(pos)
     }
 }
 
+#[cfg(unix)]
```

Comment:
Maybe forward `AsRawHandle` for `cfg(windows)` too?

Reply:
I mean, I _could_, but I was really only doing that because the helper function I used for unix used it behind the scenes.